cmd/rootlessport/wsl.go

```diff
@@ -37,3 +38,20 @@ func splitDualStackSpecIfWsl(spec rkport.Spec) []rkport.Spec {
 
 	return specs
 }
+
+func isWsl() bool {
+	if machine.HostType() == machine.Wsl {
+		return true
+	}
+
+	// "Official" way (https://github.com/Microsoft/WSL/issues/423#issuecomment-221627364)
+	content, err := os.ReadFile("/proc/sys/kernel/osrelease")
```

Comment:
Note splitDualStackSpecIfWsl() is called once per port so this is called in a hot loop over and over, I think this should be reworked to move the WSL check outside of the loop and only do it once.